joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/types/DefaultTypeInfoProvider.scala

```diff
@@ -326,56 +296,40 @@ class DefaultTypeInfoProvider(environment: KotlinCoreEnvironment) extends TypeIn
       desc = resolvedCallForSubexpression.getResultingDescriptor
     } yield desc
 
-    descMaybe.collect { case desc: FunctionDescriptor =>
-      desc
-    }
+    descMaybe.collect { case desc: FunctionDescriptor => desc }
   }
 
   def isConstructorCall(expr: KtCallExpression): Option[Boolean] = {
-    val resolvedDesc = resolvedCallDescriptor(expr)
-    resolvedDesc match {
-      case Some(fnDescriptor) =>
-        fnDescriptor match {
-          case _: ClassConstructorDescriptorImpl     => Some(true)
-          case _: TypeAliasConstructorDescriptorImpl => Some(true)
-          case _                                     => Some(false)
-        }
-      case None => None
-    }
+    resolvedCallDescriptor(expr)
+      .collect {
+        case _: ClassConstructorDescriptorImpl     => Some(true)
+        case _: TypeAliasConstructorDescriptorImpl => Some(true)
+        case _                                     => Some(false)
+      }
+      .getOrElse(None)
```

Comment:
This can be cleaned up further by having `isConstructorCall` return a `Boolean`, since it always seems to be used as
```
isConstructorCall(expr).getOrElse(false)
```

The shortest implementation I can think of is
```
resolvedCallDescriptor(expr).exists {
  case _: ClassConstructorDescriptorImpl     => Some(true)
  case _: TypeAliasConstructorDescriptorImpl => Some(true)
  case _                                     => Some(false)
}
```
which will return false if `resolvedCallDescriptor(expr)` is `None` or contains anything except the 2 matched types.

Reply:
yess